interfaces/dotnet/Cantera/src/SpeciesCollection.cs

```diff
@@ -45,16 +45,9 @@ public class SpeciesCollection : IReadOnlyList<Species>
     /// </summary>
     public double[] MassFractions
     {
-        get => InteropUtil.GetDoubles(_handle, _species.Count,
-            LibCantera.thermo_getMassFractions);
+        get => LibCantera.thermo_getMassFractions(_handle, _species.Count);
```

Comment:
The `_species.Count` information is known to `sourcegen` via `Recipe.uses` (defined in `ctthermo_auto.yaml`):
```yaml
- name: getMassFractions
  uses: nSpecies
```

While this isn't something that's visible to the user, I _believe_ that this entire `get/set` pair could be scaffolded without having hand-coded overrides.

Reply:
Ok – I have added functionality to automatically get the correct size via an interop call. Howerver, I have changed this class back to the overload that send in a span of known size – for now.